src/main/java/org/broadinstitute/hellbender/tools/exome/hmm/CopyNumberTriStateHiddenMarkovModel.java

```diff
@@ -0,0 +1,354 @@
+     */
+    private final double logQ;
+
+    /**
+     * Cached <code>log(1 - q)</code> where <code>q</code> is the <i>inverse of the mean event size in targets</i>.
+     */
+    private final double log1MinusQ;
+
+    /**
+     * Cached <code>1/D</code> where <code>D</code> is the <i>average distance between targets in the same event</i>.
+     */
+    private final double invD;
+
+    /**
+     * Deletion emission distribution.
+     */
+    private final NormalDistribution deletionDistribution;
+
+    /**
+     * Duplication emission distribution.
+     */
+    private final NormalDistribution duplicationDistribution;
+
+    /**
+     * Holds the mean number of bps between targets in the same event.
+     */
+    private final double D;
+
+    /**
+     * Holds the event-rate per target.
+     */
+    private final double P;
+
+    /**
+     * Mean number of targets per event.
+     */
+    private final double T;
+
+    /**
+     * Creates a new model instance.
+     *
+     * @param eventRate the event-rate for this instance.
+     * @param meanNumberOfTargetsPerEvent the expectation for the number of targets in an event.
+     * @param deletionMean the deletion depth of coverage negative shift.
+     * @param duplicationMean the duplication depth of coverage positive shift.
+     * @param meanEventTargetDistance the expectation of the distance between consecutive targets in an event.
+     *
+     * @throws IllegalArgumentException if any of the model parameters has an invalid value.
+     */
+    public CopyNumberTriStateHiddenMarkovModel(final double eventRate,
+                                               final double meanNumberOfTargetsPerEvent,
+                                               final double deletionMean,
+                                               final double duplicationMean,
+                                               final double meanEventTargetDistance) {
+        checkParameterRange(VALID_EVENT_RATE_RANGE, eventRate, "event rate");
+        checkParameterRange(VALID_MEAN_NUMBER_OF_TARGETS_PER_EVENT_RANGE, meanNumberOfTargetsPerEvent,
+                "mean number of targets per event");
+        checkParameterRange(VALID_DELETION_CENTER_RANGE, deletionMean, "deletion mean depth coverage shift");
+        checkParameterRange(VALID_DUPLICATION_CENTER_RANGE, duplicationMean, "duplication mean depth coverage shift");
+        checkParameterRange(VALID_MEAN_INTRA_EVENT_TARGET_DISTANCE_RANGE, meanEventTargetDistance,
+                "mean intra-event target distance");
+
+        P = eventRate;
+        logP = Math.log(eventRate);
+        log1Minus2P = Math.log1p(-2 * eventRate);
+        T = meanNumberOfTargetsPerEvent;
+        logQ = -Math.log(meanNumberOfTargetsPerEvent);
+        log1MinusQ = Math.log1p(-Math.exp(logQ));
+        D = meanEventTargetDistance;
+        invD = 1.0 / D;
+        deletionDistribution = new NormalDistribution(deletionMean, 1);
```

Comment:
Although defining `neutralDistribution` to have unit variance is definitional since we normalize to Z-scores, when `deletionDistribution` and `duplicationDistribution` are given unit variance this is a magic constant and should be defined as a static member.  The number 1.0 is an unfounded assumption of XHMM.